lib/rom/rails/model/validator/uniqueness_validator.rb

```diff
@@ -66,8 +68,8 @@ def rom
         # @return [TrueClass,FalseClass]
         #
         # @api private
-        def unique?(name, value)
-          relation.unique?(name => value)
+        def unique?(name, value, scope)
+          relation.where(scope).unique?(name => value)
```

Comment:
this should be `relation.unique?({ name => value }.merge(scope)` it'll result in the same in rom-sql and make it possible to implement that for other adapters (`where` is sql-specific interface so we can't use it here).